Response 500 Internal Server Error when failed to resolve host

Should response `502 Bad Gateway` rather than `500 Internal Server Error`.

Stacktrace:
```
com.linecorp.armeria.client.UnprocessedRequestException: java.net.UnknownHostException: failed to resolve 'berry' after 4 queries 
	at com.linecorp.armeria.client.UnprocessedRequestException.of(UnprocessedRequestException.java:45)
	at com.linecorp.armeria.client.HttpClientDelegate.finishResolve(HttpClientDelegate.java:126)
	at com.linecorp.armeria.client.HttpClientDelegate.lambda$execute$0(HttpClientDelegate.java:110)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490)
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615)
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:608)
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117)
	at com.linecorp.armeria.client.RefreshingAddressResolver.lambda$doResolve$2(RefreshingAddressResolver.java:123)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:930)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:907)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
	at com.linecorp.armeria.client.RefreshingAddressResolver.lambda$sendQueries$4(RefreshingAddressResolver.java:167)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490)
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615)
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:608)
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117)
	at io.netty.resolver.dns.DnsResolveContext.finishResolve(DnsResolveContext.java:1013)
	at io.netty.resolver.dns.DnsResolveContext.tryToFinishResolve(DnsResolveContext.java:958)
	at io.netty.resolver.dns.DnsResolveContext.query(DnsResolveContext.java:406)
	at io.netty.resolver.dns.DnsResolveContext.onResponse(DnsResolveContext.java:617)
	at io.netty.resolver.dns.DnsResolveContext.access$400(DnsResolveContext.java:63)
	at io.netty.resolver.dns.DnsResolveContext$2.operationComplete(DnsResolveContext.java:450)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:570)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:549)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490)
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615)
	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:604)
	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
	at io.netty.resolver.dns.DnsQueryContext.trySuccess(DnsQueryContext.java:201)
	at io.netty.resolver.dns.DnsQueryContext.finish(DnsQueryContext.java:193)
	at io.netty.resolver.dns.DnsNameResolver$DnsResponseHandler.channelRead(DnsNameResolver.java:1230)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:93)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.net.UnknownHostException: failed to resolve 'berry' after 4 queries 
	at io.netty.resolver.dns.DnsResolveContext.finishResolve(DnsResolveContext.java:1005)
	... 35 common frames omitted
```

--- a/core/src/main/java/dev/gihwan/tollgate/core/upstream/DefaultUpstream.java
+++ b/core/src/main/java/dev/gihwan/tollgate/core/upstream/DefaultUpstream.java
@@ -24,6 +24,9 @@
 
 package dev.gihwan.tollgate.core.upstream;
 
+import java.net.UnknownHostException;
+import java.util.concurrent.CompletableFuture;
+
 import javax.annotation.Nullable;
 
 import org.slf4j.Logger;
@@ -31,6 +34,7 @@ import org.slf4j.LoggerFactory;
 
 import com.google.common.base.Splitter;
 
+import com.linecorp.armeria.client.UnprocessedRequestException;
 import com.linecorp.armeria.common.AggregatedHttpRequest;
 import com.linecorp.armeria.common.HttpRequest;
 import com.linecorp.armeria.common.HttpResponse;
@@ -65,7 +69,7 @@ public final class DefaultUpstream implements Upstream {
 
         return HttpResponse.from(req.aggregate()
                                     .thenApply(aggregated -> buildRequest(aggregated, path))
-                                    .thenApply(service::send));
+                                    .thenApply(this::sendRequest));
     }
 
     @Nullable
@@ -94,5 +98,32 @@ public final class DefaultUpstream implements Upstream {
                                           .path(path)
                                           .build();
         return HttpRequest.of(headers, req.content(), req.trailers());
+    }
+
+    private HttpResponse sendRequest(HttpRequest req) {
+        final CompletableFuture<HttpResponse> responseFuture = new CompletableFuture<>();
+        final HttpResponse res = HttpResponse.from(responseFuture);
+        service.send(req).aggregate().handle((aggregated, t) -> {
+            if (t != null) {
+                resolveException(responseFuture, t);
+                return null;
+            }
+
+            responseFuture.complete(aggregated.toHttpResponse());
+            return null;
+        });
+        return res;
+    }
+
+    private void resolveException(CompletableFuture<HttpResponse> responseFuture, Throwable t) {
+        if (t instanceof UnprocessedRequestException) {
+            resolveException(responseFuture, ((UnprocessedRequestException) t).getCause());
+            return;
+        }
+        if (t instanceof UnknownHostException) {
+            responseFuture.complete(HttpResponse.of(HttpStatus.BAD_GATEWAY));
+            return;
+        }
+        responseFuture.completeExceptionally(t);
     }
 }

